HighestCombinationObserver/CardValidator.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using PokerHands;
+
+namespace HighestCombinationObserver
+{
+    public class CardValidator
+    {
+        public void CheckForDuplicatedCards(int gameNumber, List<Card> cards, Combination highestCombination)

```

Comment:
chodziło o sprawdzenie duplikatów nie tylko w ramach jednej ręki (co też nie dokońca tu osiągnąłeś), ale w ramach jednego rozdania (gameNumber).
A więc można to było rozwiązać np. słownikiem <int, HashSet<Card>>, który dla danego gameNumber przechowywałby zbiór użytych już w nim kart, i rzucał wyjątek w razie duplikatu.